File: advanced-programming-methods/toy-interpreter-java/src/model/statement/AssignStatement.java
package model.statement;

import model.expression.Expression;

/**
 * The type Assign statement.
 */
public class AssignStatement extends Statement
{
    private String varName;

    private Expression expression;

    /**
     * Instantiates a new Assign statement.
     *
     * @param varName the var name
     * @param expression the expression
     */
    public AssignStatement(String varName, Expression expression)
    {
        this.varName = varName;
        this.expression = expression;
    }

    /**
     * Getter for property 'expression'.
     *
     * @return Value for property 'expression'.
     */
    public Expression getExpression()
    {
        return expression;
    }

    /**
     * Getter for property 'varName'.
     *
     * @return Value for property 'varName'.
     */
    public String getVarName()
    {
        return varName;
    }

    /** {@inheritDoc} */
    @Override
    public String toString()
    {
        return String.format("%s = %s", varName, expression.toString());
    }

    /** {@inheritDoc} */
    @Override
    public Statement cloneDeep()
    {
        return new AssignStatement(varName, expression.cloneDeep());
    }
}

File: advanced-programming-methods/labs/toy-interpreter-sharp/model/statement/AssignStatement.cs
using System;

namespace Model
{
    /// <summary>
    /// The type Assign statement.
    /// </summary>
    [Serializable()]
    public class AssignStatement : Statement
    {
        private readonly string varName;

        private readonly Expression expression;

        /// <summary>
        /// Instantiates a new Assign statement.
        /// </summary>
        /// <param name="varName"> the var name </param>
        /// <param name="expression"> the expression </param>
        public AssignStatement(string varName, Expression expression)
        {
            this.varName = varName;
            this.expression = expression;
        }

        /// <summary>
        /// Getter for property 'expression'.
        /// </summary>
        /// <returns> Value for property 'expression'. </returns>
        public Expression Expression
        {
            get
            {
                return expression;
            }
        }


        /// <summary>
        /// Getter for property 'varName'.
        /// </summary>
        /// <returns> Value for property 'varName'. </returns>
        public string VarName
        {
            get
            {
                return varName;
            }
        }
            
        /// <inheritdoc />
        public override string ToString()
        {
            return string.Format("{0} = {1}", varName, expression.ToString());
        }

        /// <inheritdoc />
        public override Statement CloneDeep()
        {
            return new AssignStatement(varName, expression.CloneDeep());
        }
    }
}

